src/Nethermind/Nethermind.Merge.Plugin/Data/GetPayloadV3Result.cs

```diff
@@ -3,23 +3,26 @@
 
 using Nethermind.Core;
 using Nethermind.Int256;
+using Nethermind.Consensus.Processing;
 
 namespace Nethermind.Merge.Plugin.Data;
 
 public class GetPayloadV3Result : GetPayloadV2Result
 {
-    public GetPayloadV3Result(Block block, UInt256 blockFees, BlobsBundleV1 blobsBundle) : base(block, blockFees)
+    private readonly CensorshipDetector _censorshipDetector;
+
+    public GetPayloadV3Result(Block block, UInt256 blockFees, BlobsBundleV1 blobsBundle, CensorshipDetector censorshipDetector) : base(block, blockFees)
     {
         ExecutionPayload = new(block);
         BlobsBundle = blobsBundle;
+        _censorshipDetector = censorshipDetector;
     }
 
     public BlobsBundleV1 BlobsBundle { get; }
 
     public override ExecutionPayloadV3 ExecutionPayload { get; }
 
-    public bool ShouldOverrideBuilder { get; }
-
+    public bool ShouldOverrideBuilder => _censorshipDetector.CensorshipDetected;
```

Comment:
Make this `{get; init}` and just set it when censorship is detected.
This type is directly serialized and should be considered DTO.